Added method rep()

islas27 opened the pull request:
- Added documentation
- Added unit test

Branch based on feature/method-cat 5ea3ee1 and closes #2

islas27 committed:
Added method rep()

- Added documentation
- Added unit test

ryanez commented on README.md:
```diff
@@ -1,2 +1,77 @@
+---
+
+### rep(str, repetitions)
+The functions will append at the end of the current buffer the string sent the number of times indicates.
```
Mi no entender, what did you say?
In the other hand this is not what the requirements says about `rep` method. I'm quite sure I saw a lot more of arguments on the drive document.

islas27 replied:
Yeah, my bad. I miss understood the problem. Im working on it!

ryanez commented on test/rep.spec.js:
```diff
@@ -0,0 +1,27 @@
+    expect(result).to.equal('ones: 1')
+  })
+
+  it('convertes to a number 1 the `rep` argument if not a number', () => {
```
If you mean 'converts' you have an extra 'e' but if you meant 'convierte' your are missing an 'i' and still having an extra 's', whatever the case is, you know what I mean.

islas27 committed:
Merge branch 'master' into feature/method-rep

islas27 committed:
Fixed definition of method rep()

- Refactored code for readability
- Added test to check the correct requirements of rep()

islas27 commented:
Alright, here's the fix! @ryanez

ryanez commented on index.js:
```diff
@@ -10,10 +11,9 @@ function stringHelper () {
-      if (type === 'function') this.cat(arg.apply(this))
-      else if (type === 'array') this.cat(...arg.flatten())
-      else if (type === 'undefined' || type === 'object') continue
+      if (isType('function', arg)) this.cat(arg.apply(this))
```
using string literals among all the code is not a good practice, what about having something more semantic like
`isBoolean(value)` the linter will tell you about not defined `isBoolean` function but not that `isType('buulean', value)` will never match anything.

ryanez replied:
think about the `isNaN` function 😉

islas27 replied:
Then, if Im almost always checking if something is a function, array or boolean, should I have 3 functions (isBool, isFunc, isArr)?

islas27 replied:
BTW, tried to use `!isNaN()`, but it broke the behaviour, as a `'1'` evaluates to true, while I believe it should be treated as a string to be appended. As such, I made a `isNum()` function using `typeof`. Following this logic, made helper functions for function, array, number and elements to be ignored (like undefined and objects). Commit 94a43da

islas27 committed:
Improvements made by writting helper functions

ryanez commented:
LGTM 👍
